libs/ledgerjs/script/crypto-assets-importer/importers/currenciesExchange.js

```diff
@@ -17,14 +18,14 @@ const exchanges: Exchange[] = [
 
 export default exchanges;
 `,
-  loader: ({ signatureFolder, folder, id }) =>
-    Promise.all([
-      readFileJSON(path.join(signatureFolder, id, "exchange_signature.json")),
-    ]).then(([exchange]) => {
-      return [
-        idFromFolderAndFile(folder, id),
-        exchange.serialized_config,
-        exchange.signature,
-      ];
-    }),
+  loader: ({ signatureFolder, folder, id }) => {
+    const filePath = path.join(signatureFolder, id, "exchange_signature.json");
+    if (!fs.existsSync(filePath)) return;
+
+    return Promise.all([filePath]).then(([exchange]) => [
+      idFromFolderAndFile(folder, id),
+      exchange.serialized_config,
+      exchange.signature,
+    ]);
+  },
```

Comment:
Why we use `Promise.all` here ?

If we just want to return a promise we could just do:

```suggestion
  loader: async ({ signatureFolder, folder, id }) => {
    const filePath = path.join(signatureFolder, id, "exchange_signature.json");
    if (!fs.existsSync(filePath)) return;
    
    return  [
      idFromFolderAndFile(folder, id),
      exchange.serialized_config,
      exchange.signature,
    ];
  },
```

Or `Promise.resolve`

Reply:
You're right, I didn't even read the code that I was updating (+ I did a Promise of `filePath` instead of `ReadJSONFile(filePath)`... 🙈)

Reply:
Fixed !